src/connection.cpp

```diff
@@ -1225,8 +1225,9 @@ buffer_t* Connection::readClob(oracle::occi::Clob& clobVal, int charForm) {
     break;
   }
 
-  unsigned int lobLength = clobVal.length();
-  unsigned char* lob = new unsigned char[lobLength];
+  unsigned int clobCharCount = clobVal.length();
+  // maximum 4 bytes in a encoded character so the buffer is 4 times as large
+  unsigned char* lob = new unsigned char[clobCharCount * 4];
```

Comment:
I have not found any resources about that size for charSetId but you can probably can create your own if you want to. A list for available character sets is given though.

The [documentation](https://docs.oracle.com/database/121/LNCPP/reference009.htm#LNCPP20742) for `setCharSetId()` explains
> If no value is set explicitly, the default client's character set id is used. 

So assuming UTF8 (for V8) is used that will be 4 bytes per character max.